pkg/addonmanager/controllers/agentdeploy/controller.go

```diff
@@ -232,14 +244,54 @@ func (c *addonDeployController) updateAddon(ctx context.Context, new, old *addon
 		return err
 	}
 
-	return utils.PatchAddonCondition(ctx, c.addonClient, new, old)
+	if equality.Semantic.DeepEqual(new.Status.HealthCheck, old.Status.HealthCheck) &&
+		equality.Semantic.DeepEqual(new.Status.Conditions, old.Status.Conditions) {
+		return nil
+	}
+
+	oldData, err := json.Marshal(&addonapiv1alpha1.ManagedClusterAddOn{
+		Status: addonapiv1alpha1.ManagedClusterAddOnStatus{
+			HealthCheck: old.Status.HealthCheck,
+			Conditions:  old.Status.Conditions,
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	newData, err := json.Marshal(&addonapiv1alpha1.ManagedClusterAddOn{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:             new.UID,
+			ResourceVersion: new.ResourceVersion,
+		},
+		Status: addonapiv1alpha1.ManagedClusterAddOnStatus{
+			HealthCheck: new.Status.HealthCheck,
+			Conditions:  new.Status.Conditions,
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	patchBytes, err := jsonpatch.CreateMergePatch(oldData, newData)
+	if err != nil {
+		return fmt.Errorf("failed to create patch for addon %s: %w", new.Name, err)
+	}
+
+	klog.V(2).Infof("Patching addon %s/%s condition with %s", new.Namespace, new.Name, string(patchBytes))
+	_, err = c.addonClient.AddonV1alpha1().ManagedClusterAddOns(new.Namespace).Patch(
+		ctx, new.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, "status")
+	return err
 }
 
 func (c *addonDeployController) applyWork(ctx context.Context, appliedType string,
 	work *workapiv1.ManifestWork, addon *addonapiv1alpha1.ManagedClusterAddOn) (*workapiv1.ManifestWork, error) {
 
 	work, err := c.workApplier.Apply(ctx, work)
 	if err != nil {
+		if errors.IsAlreadyExists(err) {
```

Comment:
will this return "nil, nil"? should we change this in applier instead of here?

Reply:
removed the check ,added into applier